### ISSUE TITLE
Uyuni Bugreport - Uyuni 2021.06: Insesure TLSv1.0 and TLSv.11 still activated on Uyuni proxies

### ISSUE THREAD

**Issue by phibid** [uyuni-project/uyuni#4105]
Although TLSv.1.0 and 1.1 are deactivated in the Uyuni server, this is not the case for the Uyuni proxies.

I think that this line in the default SSL vhost should be removed in the file `/etc/apache2/vhosts.d/ssl.conf`:

```
SSLProtocol all -SSLv2 -SSLv3
```

As the SSL configuration is correct in conf file `/etc/apache2/ssl-global.conf`

```
        SSLProtocol all -SSLv3 -TLSv1 -TLSv1.1
```

However it is not taken into account (even after removing the SSLProtocol configuration in `vhosts.d/ssl.conf`), not sure why as it is working on the Uyuni server and TLSv1.0 and 1.1 are enabled.

**Comment by Shirocco88** [comment]
@phibid :
what is the output of "apachectl -S" @uyuni-srv vs. @uyuni-proxy

**Comment by phibid** [comment]
Server:

```
# apache2ctl -S
[Fri Aug 06 15:33:55.026999 2021] [so:warn] [pid 40841] AH01574: module wsgi_module is already loaded, skipping
[Fri Aug 06 15:33:55.027078 2021] [so:warn] [pid 40841] AH01574: module proxy_module is already loaded, skipping
[Fri Aug 06 15:33:55.027088 2021] [so:warn] [pid 40841] AH01574: module proxy_http_module is already loaded, skipping
VirtualHost configuration:
*:443                  uyuni (/etc/apache2/vhosts.d/vhost-ssl.conf:25)
ServerRoot: "/srv/www"
Main DocumentRoot: "/srv/www/htdocs"
Main ErrorLog: "/var/log/apache2/error_log"
Mutex default: dir="/run/" mechanism=default 
Mutex mpm-accept: using_defaults
Mutex ssl-stapling-refresh: using_defaults
Mutex rewrite-map: using_defaults
Mutex ssl-stapling: using_defaults
Mutex proxy: using_defaults
Mutex ssl-cache: using_defaults
PidFile: "/var/run/httpd.pid"
Define: SYSCONFIG
Define: SSL
Define: ISSUSE
Define: DUMP_VHOSTS
Define: DUMP_RUN_CFG
User: name="wwwrun" id=473
Group: name="www" id=476
```

Proxy:

```
# apache2ctl -S
VirtualHost configuration:
*:443                  proxy (/etc/apache2/vhosts.d/ssl.conf:25)
ServerRoot: "/srv/www"
Main DocumentRoot: "/srv/www/htdocs"
Main ErrorLog: "/var/log/apache2/error_log"
Mutex default: dir="/run/" mechanism=default 
Mutex mpm-accept: using_defaults
Mutex ssl-stapling-refresh: using_defaults
Mutex rewrite-map: using_defaults
Mutex ssl-stapling: using_defaults
Mutex proxy: using_defaults
Mutex ssl-cache: using_defaults
PidFile: "/var/run/httpd.pid"
Define: SYSCONFIG
Define: SSL
Define: DUMP_VHOSTS
Define: DUMP_RUN_CFG
User: name="wwwrun" id=473
Group: name="www" id=476
```

**Comment by Shirocco88** [comment]
I've checked the source code and found TLS1.1 / TLSv1.2 code
https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/common/util/http/HttpClientAdapter.java

    /**
     * Initialize an {@link HttpClient} for performing requests. Proxy settings will
     * be read from the configuration and applied transparently.
     */
    public HttpClientAdapter() {
        Optional<SSLConnectionSocketFactory> sslSocketFactory = Optional.empty();
        try {
            SSLContext sslContext = SSLContext.getDefault();
            sslSocketFactory = Optional.of(new SSLConnectionSocketFactory(
                    sslContext,
                    new String[]{"TLSv1", "TLSv1.1", "TLSv1.2"},
                    null,
                    SSLConnectionSocketFactory.getDefaultHostnameVerifier()));
        }

**Comment by phibid** [comment]
So if I understand correctly your last post, you do not recommend the deactivation of TLS1.0 and TLS1.1 on the Uyuni proxies ?

**Comment by mcalmer** [comment]
Should be fixed in uyuni 2021.10
